pyo3-build-config/src/impl_.rs

```diff
@@ -219,7 +219,12 @@ print("mingw", get_platform() == "mingw")
         let implementation = map["implementation"].parse()?;
 
         let lib_name = if cfg!(windows) {
-            default_lib_name_windows(version, abi3, map["mingw"].as_str() == "True")
+            default_lib_name_windows(
+                version,
+                implementation,
+                abi3,
+                map["mingw"].as_str() == "True",
```

Comment:
According to https://github.com/PyO3/maturin/issues/676#issuecomment-967083567 , `sysconfig.get_platform()` returns `mingw_x86_64` instead of just `mingw` in L203.